File: w03/d05/liza_ramo/Morning Exercise Calculator/lib/calculator.rb

class Calculator

  def add(num1, num2)
    num1 + num2
  end

  def subtract(num1, num2)
    num1 - num2
  end

  def sum(array)
    if array == []
      return 0
    else
      array.reduce(:+)
    end
  end

  def multiply(*args)
    args.reduce(:*)
  end

  def power(number_one, number_two)
    if number_two == 0
      return 1
    else
      number_one ** number_two
    end
  end

  def factorial(num)
    num == 0 ? 1 : (1..num).reduce(:*)
  end

end

File: w06/d05/kathryn_inman/calculator.js
// Add
var add = function(num1, num2) {
	return num1 + num2;
};

console.log("addition result: " + add(2, 3));

// Subtract
var subtract = function(num1, num2) {
	return num1 - num2;
};

console.log("subtraction result: " + subtract(5, 2));

// Times (num1 multiplied by aka times num2)
var times = function(num1, num2) {
	var answer = 0;
	for (var integer = 0; integer < num2; integer++) {
		answer += num1;
	}
	return answer;
};

console.log("times result: " + times(6, 5));

// Divided (num1 divided by num2)
var dividedBy = function(numerator, denominator) {
	var counter = 0;
	while (numerator >= denominator){
		numerator -= denominator;
		counter = counter + 1;
	}
	return counter;	
};


console.log("division result: " + dividedBy(12, 5));

// Factorial 
var factorial = function(num) {
	var result 
}